### PR TITLE
Update to Ingest Data into Delta Lake Using Azure Synapse

### PR DESCRIPTION
**Pull request by kingfardthe1st**
Modeled after the `eventhubs-capture-databricks-delta` example, this update adds an end-to-end example to demonstrate ingesting and processing EventHub captured data into an Azure Storage Delta Lake using Azure Synapse as the processing layer. 

Please refer to the README for more details on the solution and how to run the script.

### CODE REVIEW

**Comment by benjguin** [comment]
I think the title of the PR should be changed. There doesn't seem to be such a thing as Azure storage Delta Lake. It's an implementation of Apache Delta Lake inside Azure Synapse. See <https://docs.microsoft.com/en-us/azure/synapse-analytics/spark/apache-spark-what-is-delta-lake>.

File: eventhubs-capture-synapse-delta/README.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    az account list
+
+if you want to select a specific subscription use the following command
+
+    az account set --subscription <subscription_name>
+
+once you have selected the subscription you want to use just execute the following command
+
+    ./create-solution.sh -d <solution_name> -p <synapse_sql_password> -w <wait_verfication>
```
Comment (algattik) [inline eventhubs-capture-synapse-delta/README.md:57]: how about managing the password similarly to eventhubs-streamanalytics-azuresql/create-solution.sh line 186? There are various password complexity rules for the various services, and users can be confused with them.

File: eventhubs-capture-synapse-delta/create-solution.sh [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+export SYNAPSE_SPARKPOOL="sasesssparkpool"
+
+export PREFIX=''
+export LOCATION="eastus"
+export TESTTYPE="1"
+export STEPS="CIPTMV"
+export WAITVERIFICATION=true
+
+usage() {
+    echo "Usage: $0 -d <deployment-name> $1 -p <sparkpool-sql-password> [-s <steps>] [-t <test-type>] [-l <location>] [-w <wait-verfication>]"
```
Comment (algattik) [inline eventhubs-capture-synapse-delta/create-solution.sh:29]: please document this `-w` parameter. (Though my preference would be to remove it completely, for simplicity, the user can just cancel the script execution)

File: eventhubs-capture-synapse-delta/create-solution.sh [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    echo "-s: specify which steps should be executed. Default=$STEPS"
+    echo "    Possible values:"
+    echo "      C=COMMON"
+    echo "      I=INGESTION"
+    echo "      P=PROCESSING"
+    echo "      T=TEST clients"
+    echo "      M=METRICS reporting"
+    echo "      V=VERIFY deployment"
+    echo "-t: test 1,5,10 thousands msgs/sec. Default=$TESTTYPE"
+    echo "-l: where to create the resources. Default=$LOCATION"
```
Comment (algattik) [inline eventhubs-capture-synapse-delta/create-solution.sh:39]: doc should match command line parameters

File: eventhubs-capture-synapse-delta/create-solution.sh [hunk 4]
```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+
+# Strict mode, fail on any error
+set -euo pipefail
+source set-variables
+source ../streaming/synapse/job/run-synapse-pipeline.sh
```
Comment (algattik) [inline eventhubs-capture-synapse-delta/create-solution.sh:6]: running this without arguments yields an unfriendly message

```
Enter a name for this deployment.
./create-solution.sh: line 29: $1: unbound variable
```

File: eventhubs-capture-synapse-delta/create-solution.sh [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+on_error() {
+    set +e
+    echo "There was an error, execution halted" >&2
+    echo "Error at line $1"
+    exit 1
+}
+
+trap 'on_error $LINENO' ERR
+
+export SQL_ADMIN_USER="sasesssynadmin"
```
Comment (algattik) [inline eventhubs-capture-synapse-delta/create-solution.sh:17]: why these strange values? 

In create-sql.sh we hard-code the admin user (for simplicity) to the value `serveradmin`. Is there a reason to do things differently here?

File: eventhubs-capture-synapse-delta/create-solution.sh [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+echo "***** [P] Setting up PROCESSING"
+
+    export ADB_WORKSPACE=$PREFIX"synapse"
+    export ADB_TOKEN_KEYVAULT=$PREFIX"kv" #NB AKV names are limited to 24 characters
+    export BLOB_FILE_FORMAT="avro"
+
+    RUN=`echo $STEPS | grep P -o || true`
+    if [ ! -z "$RUN" ]; then
+        echo "Setting up processing. Currently there is no processing layer."
+        source ../components/azure-synapse/create-synapse.sh $SQLPASSWORD
```
Comment (algattik) [inline eventhubs-capture-synapse-delta/create-solution.sh:175]: Getting worrying warning messages:

```
***** [P] Setting up PROCESSING
Setting up processing. Currently there is no processing layer.
Creating Azure Synapse Workspace agsynapse-synwkspc
Parameter --repository-type is missing, the following arguments are ignored: --host-name ,--account-name ,--collaboration-branch ,--repository-name ,--root-folder--project-name ,--tenant-id. Repository configuration will not work.
```

Also, JSON output should be silenced with `-o none`:
```
{
  "adlaResourceId": null,
  "azureAdOnlyAuthentication": null,
  "connectivityEndpoints": {
```

File: components/azure-synapse/create-synapse.sh [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  # Creating a notebook requires a firewall rule to allow access to Azure Synapse Workspace from your machine
+  WorkspaceWeb=$(az synapse workspace show --name $SYNAPSE_WORKSPACE --resource-group $RESOURCE_GROUP | jq -r '.connectivityEndpoints | .web')
+  WorkspaceDev=$(az synapse workspace show --name $SYNAPSE_WORKSPACE --resource-group $RESOURCE_GROUP | jq -r '.connectivityEndpoints | .dev')
+  ClientIP=$(curl -sb -H "Accept: application/json" "$WorkspaceDev" | jq -r '.message')
+  ClientIP=${ClientIP##'Client Ip address : '}
+
+  echo "Creating a firewall rule to enable access for IP address: $ClientIP"
+  FIREWALL_RULE_NAME="Allow Client IP"
+
+  az synapse workspace firewall-rule create --end-ip-address $ClientIP \
```
Comment (algattik) [inline components/azure-synapse/create-synapse.sh:40]: is this necessary? NB the Azure SQL DB template doesn't do this

File: eventhubs-capture-synapse-delta/create-solution.sh [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+        source ../components/azure-event-hubs/report-throughput.sh
+    fi
+echo
+
+echo "***** [V] Starting deployment VERIFICATION"
+
+    RUN=`echo $STEPS | grep V -o || true`
+    if [ ! -z "$RUN" ]; then
+        source ../components/azure-synapse/create-synapse.sh $SQLPASSWORD
+        source ../streaming/synapse/runners/verify-delta.sh $WAITVERIFICATION
```
Comment (algattik) [inline eventhubs-capture-synapse-delta/create-solution.sh:202]: it seems the script is not setting its exit code properly if this fails?

**Review by algattik — COMMENTED** [review]
comments provided

**Comment by algattik** [comment]
Please reopen after addressing comments.

**Pull request closed without merging**